lib/thrift/servers/binary_framed/server.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Thrift.Servers.BinaryFramed do
+  @moduledoc false
+
+  @type server_opts :: [
+    worker_count: pos_integer
+  ]
+
+  alias Thrift.Servers.BinaryFramed.AcceptorSupervisor
+
+  defmodule State do
+    @moduledoc false
+    defstruct port: nil, socket: nil, supervisor: nil
+  end
+
+  @type opts :: [
+    port: (1..65535),
+  ]
+
+  @spec start_link(module, (1..65535), module, server_opts, GenServer.option) :: GenServer.on_start
+  def start_link(server_module, port, handler_module, opts, gen_server_opts) do
+    GenServer.start_link(__MODULE__, [server_module, port, handler_module, opts], gen_server_opts)
+  end
+
+  def init([server_module, port, handler_module, opts]) do
+    worker_count = Keyword.get(opts, :worker_count, 1)
+
+    {:ok, socket} = :gen_tcp.listen(port, [:binary, packet: 4, active: false])
+    {:ok, supervisor} = AcceptorSupervisor.start_link(socket, server_module, handler_module, worker_count)
```

Comment:
This supervisor is not part of a supervision tree, and it's linked to the server. If the server dies, it will kill all the connections as well. And when the whole application shuts down, it won't get a chance to close the connections cleanly (for instance by finishing the current request and not starting another). I wonder if we can set up a structure such that server process (holding the listen socket) and the connection processes (holding the regular sockets) are supervised independently. Maybe a topic for a future PR though.